server/app/api/v3.py

```diff
@@ -85,10 +101,60 @@ def get(self, data_adaptor):
 
 
 class AnnotationsObsAPI(S3URIResource):
-    @cache_control(immutable=True, max_age=ONE_YEAR)
     @rest_get_s3uri_data_adaptor
     def get(self, data_adaptor):
-        return common_rest.annotations_obs_get(request, data_adaptor)
+        # Check if request includes user annotations to determine caching
```

Comment:
annotations/obs GET now sets Cache-Control based on whether user annotations are involved: immutable for built-ins, no-store when personalized data is present.